internal/pkg/model/filter.go

```diff
@@ -20,30 +20,73 @@ type AllowedBranches []AllowedBranch
 
 type ComponentIds []ComponentId
 
-type Filter struct {
-	AllowedBranches   AllowedBranches `json:"allowedBranches" validate:"required,min=1"`
-	IgnoredComponents ComponentIds    `json:"ignoredComponents"`
+// ObjectsFilter filters objects by allowed keys, allowed branches and ignored components.
+type ObjectsFilter struct {
```

Comment:
Pri vytvarani template user vybera, ktore objekty maju byt sucastou template.
Potrebujeme teda filtrovat objekty pri nacitani, ... to uz ciastocne mame, riesi to `model.Filter`, ktory filtruje `AllowedBranches` a `IgnoredComponents` ... v tomto PR ho rozsirujem.

Pridavam `allowedKeys`, aby bolo mozne filtrovat jednotlive objekty.

Reply:
Tam se budou z nějakýho důvodu ukládat i objekty s hodnotou `false`, nebo je to tak jen proto aby se to líp používalo? (než řekněme ekvivalent phpkovýho `in_array()`. Hádám, že by se to pole v Go muselo proiterovat, co?)

Reply:
jj presne tak, to ma zlozitost `O(1)`. ako nie ze by na tom nejak zalezalo, ale je to aj kratsi kod.

z map sa da citat ako `value, ok := m[...]` alebo `value := m[..]`

tu sa pouziva ten 2 pripad, ... tj. nerozlisujeme medzi nenastavenou a `false` hodnotou, v oboch pripadoch to vrati `false` ... a tak to je ekvivalent `in_array`, len to ma zlozitost `O(n)`.